drivers/display/display_sdl.c

```diff
@@ -288,11 +371,16 @@ static int sdl_display_write(const struct device *dev, const uint16_t x,
 	} else if (disp_data->current_pixel_format == PIXEL_FORMAT_L_8) {
 		sdl_display_write_l8(disp_data->buf, desc, buf);
 	}
+	k_mutex_unlock(&disp_data->task_mutex);
```

Comment:
Shouldn't this mutex remain locked all the way until the end of the function to also protect the read in the direct call to `exec_sdl_task()`?